interface/wx/textctrl.h

```diff
@@ -1761,6 +1761,35 @@ class wxTextCtrl : public wxControl,
     */
     virtual long XYToPosition(long x, long y) const;
 
+    /**
+        @name Mac-specific functions
+    */
+    //@{
+
+    /**
+      Enable the automatic replacement of new lines characters in a single-line text field
+      with spaces under MacOS.
+
+      This feature is enabled by default and will replace any new line (\n) character entered
+      into a single-line text field with the space character.
+      Usually single-line text fields are not expected to hold multiple lines of text (that is
+      what wxTE_MULTILINE is for, after all). However, under MacOS/Cocoa, a multi-line text control
+      created with the wxTE_MULTILINE flag will not respond correctly to keypressed when run from
+      inside a plugin.
+      In that case, this function can be called on a single-line text control (created without the
+      wxTE_MULTILINE flag) to disable the new line replacement and turn it into a multi-line text
+      control that also works correctly from inside a plugin.
```

Comment:
I'm not sure if we really want to describe this situation (which I still think is due to a bug in our code) as motivation for having the function, I think we could just say something like this instead:
```suggestion
      what wxTE_MULTILINE is for, after all) and it is impossible to have multiple
      lines of text in them under non-Mac platforms. However, under MacOS/Cocoa, a
      single-line text control can still show multiple lines and this function allows
      to lift the restriction preventing multiple lines from being entered unless
      wxTE_MULTILINE is specified.
```